api/rosetta/balance.go

```diff
@@ -51,37 +51,21 @@ func (d *Data) Balance(ctx echo.Context) error {
 		return echo.NewHTTPError(http.StatusBadRequest, invalidEncoding(invalidJSON, err))
 	}
 
-	if req.NetworkID.Blockchain == "" {
-		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(blockchainEmpty))
-	}
-	if req.NetworkID.Network == "" {
-		return echo.NewHTTPError(http.StatusBadRequest, invalidFormat(networkEmpty))
-	}
-
-	if req.BlockID.Hash != "" && len(req.BlockID.Hash) != hexIDSize {
+	err = d.Validate(req)
+	if errors.Is(err, errBlockLength) {
```

Comment:
This special casing for errors is only needed when we want to add case-specific parameters.

Reply:
Why not have this:

```go
err = d.Validate(req)
if err != nil {
  return rosettaError(err)
}
```

Then you can handle all of the special sentinel error cases in a single function.

Reply:
I was thinking of that, and had something similar in one of the previous iterations.

But, if we make it generic, we lose the connection to the original data that caused the error. We can't derive the `have_length` and `want_length` from the error alone*.

However, I think having something like:

```go
var req BalanceRequest
err := d.Unpack(ctx, &req)
if err != nil {
	return rosettaError(err)
}
```
Where `Unpack` does `Bind()` and then `Validate()`.
________________________________
*- Actually we can have the `want_length` from the error, in a roundabout, tad ugly way. Validator library allows us to set one parameter for an error. That could be the `want_length` parameter. Downside - it has to be a string